src/qt/bitcoingui.cpp

```diff
@@ -441,6 +441,42 @@ void BitcoinGUI::createActions()
     }
 #endif // ENABLE_WALLET
 
+    m_chain_selection_action = new QAction(tr("Switch &chain"), this);
+    m_chain_selection_action->setStatusTip(tr("Restart application using a different (test) network"));
+    m_chain_selection_menu = new QMenu(this);
+
+    connect(m_chain_selection_menu, &QMenu::aboutToShow, [this] {
+        m_chain_selection_menu->clear();
+        const std::vector<std::pair<QString, QString>> chains = {{"main", "&Bitcoin"}, {"test", "&Testnet"}, {"regtest", "&Regtest"}, {"signet", "&Signet"}};
+        for (auto chain : chains) {
+            QAction* action = m_chain_selection_menu->addAction(chain.second);
+
+            if (gArgs.GetChainName() == chain.first.toStdString()) {
+              action->setEnabled(false);
```

Comment:
nit, fix indentation heer and below.

Reply:
nit, could do `action->setEnabled(gArgs.GetChainName() != chain.first.toStdString())`.